lib/Serialization/Deserialization.cpp

```diff
@@ -2612,9 +2612,6 @@ ModuleFile::getDeclChecked(DeclID DID, Optional<DeclContext *> ForcedContext) {
       return nullptr;
     }
 
-    if (auto interfaceType = getType(interfaceTypeID))
```

Comment:
Where does the underlying type actually get set though? We can't rely on validateDecl() doing the right thing with deserialized decls.